### DISCUSSION TITLE
aula JWT Claims, curso Fundamentos ASP.NET 6, dúvida: Identity

### DISCUSSION THREAD

**Discussion by dhionys-soares** [balta-io/forum#533] [Tira Dúvidas] [answered]
Olá, não consigo entender o Identity.
Como que ele está disponível para uso através do User.Identity.IsAuthenticated se o Identity não foi configurado?.. Ele está dentro da configuração do Token?

**Comment by henriqueweege** [comment, marked as answer]
Oi, @dhionyssoares ! Tudo certo com você?

O "User" é uma propriedade do tipo "ClaimsPrincipal" que está dento da classe  "ControllerBase", que a gente herda quando cria uma nova controller. 
![image](https://github.com/balta-io/forum/assets/96916801/9499d316-d3cd-42fe-a7f3-5129bf0d3cc7)

Se olharmos dentro dessa classe ("ClaimsPrincipal"), vemos que existe uma propriedade "Identity", que é do tipo "IIdentity".
![image](https://github.com/balta-io/forum/assets/96916801/f24dd2d6-2b89-444e-94b8-ffb8b9692663)

Se olharmos dentro dessa interface ("IIdentity"), vemos que existem algumas propriedades, entre elas, o "IsAuthenticated".
![image](https://github.com/balta-io/forum/assets/96916801/751f873f-bb90-41b6-99ff-9d94772a44ff)


Isso quer dizer que quando é criada uma instância da nossa controller, o .NET se vira para montar o objeto a partir do Token da requisição. No caso do "IsAuthenticated", dependendo do resultado da verificação que é feita pelo framework a propriedade vai estar setada de um jeito ou de outro.

Valeeeu! Feliz Páscoa!
